Publish: Merge develop to master to publish the docs.

neoblizz opened the pull request:
We can figure out a better way to do this, but for now, I am using develop as a branch I edit on and master as the live version. And when a Pull Request is merge from develop to master, you make the changes live.

bkj commented:
- Drafts for SGM, VN, GraphSearch, and GraphProjections here.
- Still need to start ApplicationClassification and LocalGraphClustering

jowens commented on source/hive_graphsearch.html.md:
````diff
@@ -0,0 +1,272 @@
+```
+git clone --recursive https://github.com/gunrock/gunrock -b dev-refactor
+cd gunrock/tests/rw/
+cp ../../gunrock/util/gitsha1.c.in ../../gunrock/util/gitsha1.c
````
This seems weird ... I thought this was in the makefile somehow? We shouldn't have to do this manually.

bkj replied:
I agree.  Easy to fix, but I wasn't sure what the `../../gunrock/util/gitsha1.c` file was supposed to be.

jowens commented on source/hive_graphsearch.html.md:
````diff
@@ -0,0 +1,512 @@
+The scoring model can be an arbitrary function, eg of node metadata.  For example, if we were running `GraphSearch` on the Twitter friends/followers graph, the scoring model might be the output of a text classifier on each users' messages.  Thus, we do not implement the scoring model in our Gunrock implementation -- we read scores from an input file and access them as necessary.
+
+`GraphSearch` is a generalization of a random walk algorithm, where there can be more variety in the transition function between nodes.  The `GraphSearch` `uniform` case is exactly a uniform random walk, so we can use the pre-existing Gunrock application.  Given a node, we compute the node to walk to as:
+```python
````
It is perhaps useful to also express this base algorithm in terms of Gunrock primitives?

jowens commented on source/hive_graphsearch.html.md:
```diff
@@ -0,0 +1,512 @@
+
+### Performance limitations
+
+<TODO>
```
Almost certainly this is random memory access (every vertex reads a random vertex). Might be useful to compute the memory bandwidth that we are sustaining given this assumption.

jowens commented on source/hive_graphsearch.html.md:
```diff
@@ -0,0 +1,512 @@
+We measure runtime on the [HIVE graphsearch Twitter dataset](https://hiveprogram.com/data/_v0/graph_search/).
+ - Nodes: 9291392
+ - Edges: 21741663
+
```
Perhaps do a one-sentence/-table performance summary here to show how Gunrock is doing compared to the reference implementations.

bkj replied:
added

jowens commented on source/hive_graphsearch.html.md:
```diff
@@ -0,0 +1,512 @@
+
+### Gunrock implications
+
+For the `greedy` and `stochastic_greedy` transition function, we have to sequentially iterate over all of a node's neighbors.  Simple wrappers for computing eg. the maximum of node scores across all of a nodes neighbors could be helpful, both for ease of programming and performance.
```
@sgpyc is this something that can be encapsulated in the neighborreduce kernel?

jowens commented on source/hive_graphsearch.html.md:
```diff
@@ -0,0 +1,512 @@
+
+### Notes on larger datasets
+
+<TODO>
```
Check with @sgpyc here -- my general understanding is that (at a high level) if inter-GPU communication is proportional to `|V|`, we're generally going to scale well, and proportional to `|E|`, we won't. This application should require communication proportional to `|V|`, by my reading.

jowens commented on source/hive_proj.html.md:
```diff
@@ -0,0 +1,332 @@
+The planned implementation of additional semirings in GraphBLAS would extend the kinds of weightings we could support. 
+
+### Comparison against existing implementations
+
```
Please do a two-sentence performance summary of our-implementation vs. other-implementations.

jowens commented on source/hive_proj.html.md:
```diff
@@ -0,0 +1,332 @@
+
+Another straightforward way to implement graph projections would be as a single sparse matrix multiplication `G.T.dot(G)` -- it would be interesting to compare the performance of this Gunrock implementation with a high-quality GPU SpMM-based implementation.
+
+As mentioned above, it would also be worthwhile to implement a version that does not require allocating the `|V|x|V|` array.  This could be accomplished by using a sparse data structure (eg, a hashmap or a mutable graph), or possibly by moving intermediate results from GPU to CPU memory during the computation.
```
Dynamic (mutable) data structures on GPUs are a significant research challenge. For our info, can you be specific about what operations this data structure must support? e.g.: "many parallel nodes call `insert(key, value)` with the following semantics if `key` is already present", "many parallel threads call `value = lookup(key)`", etc.

jowens commented on source/hive_proj.html.md:
```diff
@@ -0,0 +1,332 @@
+
+### Gunrock implications
+
+Gunrock support for bipartite graphs would make programming bipartite graph projections easier and more scalable.
```
Please be more specific. What specific support would be useful?

jowens commented on source/hive_sgm.html.md:
```diff
@@ -0,0 +1,188 @@
+ - To penalize adjacency disagreements, they convert the 0s in the inputs matrices `A` and `B` to -1s
+ - They initialize the solution matrix `P` to the barycenter of the polytope of doubly-stochastic matrices, which has all entries != 0.
+
+In order to take advantage of sparsity and make SGM more suitable for HIVE analysis, we make two small modifications to the algorithm:
```
Be specific here: this is what we did for the Gunrock implementation only? Did the OpenMP reference implementations do the same? It would probably be helpful for all readers if you were specific about how these changes allow us to take advantage of sparsity.

jowens replied:
Also a sentence on whether you expect these changes make it more? less? accurate (if you have any idea at all, which I know is really hard on this workflow).

bkj commented:
@ctcyang  Can you address John's comments above that I tagged with a ❤️ ?

neoblizz commented on source/hive_vn.html.md:
```diff
@@ -0,0 +1,382 @@
+
+### Notes on larger datasets
+
+__TODO__
```
If the datasets are larger than a single or multi-GPU's aggregate memory, the straightforward solution would be to let Unified Virtual Memory (UVM) in CUDA automatically handle memory movement. Declaring the entire graph as `managed` memory for a single GPU implementation, will allow the users to simply oversubscribe for 1 GPU, and queue vertices and edges in from the host memory as needed (this will be very slow). Further optimizations can be done, where instead of utilizing host memory, we can leverage a multi-GPU implementation and move the entire graph over to device memory, using NVLink to move data between the devices' global memory. This can be further optimized by using MemAdvise hints such as pinning the memory to GPU's local memory where it is likely to be used, or create a direct map to all other GPUs to avoid page faulting on first touch.

snyk-bot committed:
fix: upgrade docsify from 4.10.2 to 4.11.2

Snyk has created this PR to upgrade docsify from 4.10.2 to 4.11.2.

See this package in NPM:
https://www.npmjs.com/package/docsify

See this project in Snyk:
https://app.snyk.io/org/anikethsaha/project/6efe7e04-4e52-497e-8ad5-570733cc577a?utm_source=github&utm_medium=upgrade-pr

anikethsaha committed:
Merge pull request #1099 from docsifyjs/snyk-upgrade-82b6c6427661ce914c39a1be1464a76e

[Snyk] Upgrade docsify from 4.10.2 to 4.11.2

Koooooo-7 committed:
[fix #1115] Image resize in % doesn't work.

anikethsaha committed:
Merge pull request #1116 from docsifyjs/fix1115

[fix #1115] Image resize in % doesn't work.

sy-records committed:
fix: Inconsistent search and body rendering

* fix

* Update search.js

* Update search.js

* Update search.js

* Update search.js

* Update search.js

trusktr committed:
Update nameLink option docs (#1125)

sy-records committed:
Update index.html (#1144)

vagra committed:
reset and modify.

sytone committed:
Add front matter to dev page
Update embeded files documentation
Add example with front matter
Add various comments I used to understand code
Updated front matter to have installed flag
Added function to parse front matter
Added intercept to remove front matter for embed

Commit:
Improve accessibility of viewport meta

sy-records committed:
Modify Chinese doc link (#1151)

sy-records committed:
fix Edit Document

anikethsaha committed:
Merge pull request #1146 from vagra/bugfix

Fix #1136 and #1127

emersonreisdev committed:
Fix typo on `externalLinkRel`

Also, add a link explaining why that can be a security issue.

anikethsaha committed:
Merge pull request #1156 from dungahk/patch-1

Fix typo on `externalLinkRel`

anikethsaha committed:
Merge pull request #1149 from chicgeek/feature/viewport-meta-accessibility

Improve accessibility of viewport meta by allowing page zoom (related to #945)

anikethsaha committed:
Merge pull request #1148 from sytone/develop

Updated embed to strip YAML front matter fixes #1129

anikethsaha committed:
Merge pull request #1152 from sy-records/fix-edit-on-github

fix Edit Document

neoblizz committed:
:bookmark: Moving gunrock documentation to docsify.
